test cases/common/215 link custom/meson.build

```diff
@@ -57,3 +57,15 @@ d2_i = declare_dependency(link_whole: clib[0])
 
 exe4_i = executable('prog4_i', 'prog.c', dependencies: d2_i)
 test('linkwhole2_i', exe2_i)
+
+# Link with custom target
+
+py_mod = import('python')
+py3 = py_mod.find_installation()
+t = custom_target('dummy', output: [ 'libdummy.a' ], command: [ py3, '--version' ])
```

Comment:
This is not testing the behaviour properly. The custom target does not actually seem to generate a library so the link operations below are useless and in fact they should fail. It's strange that they don't. A proper test should do something like:

- build a static library called dummy which has a function `inner_lib_func` or something
- create a custom target that takes the dummy lib as input and creates a different lib, say `libcustom` by copying the input lib file to the output lib file
- the static library `lib1` should have a function `outer_lib_func` that calls `inner_lib_func`
- there should be a test binary `exe1` that links against `lib1` (and only that) and calls `outer_lib_func` and there should be a test that runs this
- there should also be a `exe2` that does the same for `lib2`

Reply:
Updated following your advice. As is noted in common test case 28, it's not possible to emulate `cp` on windows so it will be skipped in case meson can't find it.

Reply:
Use of shell commands in Meson is forbidden Write a Python script that does the copying instead. It works on all platforms.

Reply:
Updated.